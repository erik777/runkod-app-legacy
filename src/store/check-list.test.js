import reducer, {
  checkListAddAct,
  checkListDeleteAct,
  checkListResetAct
} from "./check-list"


let state = undefined;

it('1- default state', () => {
  expect(reducer(state, {})).toMatchSnapshot();
});

it('2- add ', () => {
  const act = checkListAddAct("foo");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('3- add ', () => {
  const act = checkListAddAct("bar");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('4- add ', () => {
  const act = checkListAddAct("baz");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('4- remove ', () => {
  const act = checkListDeleteAct("bar");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});


it('5- remove ', () => {
  const act = checkListDeleteAct("foo");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('6- remove ', () => {
  const act = checkListDeleteAct("baz");
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('7 add ', () => {
  const act = checkListAddAct(["bar", "baz"]);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('8 reset ', () => {
  const act = checkListResetAct();
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('9 add ', () => {
  const act = checkListAddAct(["bar", "baz"]);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});

it('10 add ', () => {
  const act = checkListAddAct(["baz"]);
  state = reducer(state, act);
  expect(state).toMatchSnapshot();
});