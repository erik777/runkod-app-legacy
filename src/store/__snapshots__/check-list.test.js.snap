// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1- default state 1`] = `Array []`;

exports[`2- add  1`] = `
Array [
  "foo",
]
`;

exports[`3- add  1`] = `
Array [
  "foo",
  "bar",
]
`;

exports[`4- add  1`] = `
Array [
  "foo",
  "bar",
  "baz",
]
`;

exports[`4- remove  1`] = `
Array [
  "foo",
  "baz",
]
`;

exports[`5- remove  1`] = `
Array [
  "baz",
]
`;

exports[`6- remove  1`] = `Array []`;

exports[`7 add  1`] = `
Array [
  "bar",
  "baz",
]
`;

exports[`8 reset  1`] = `Array []`;

exports[`9 add  1`] = `
Array [
  "bar",
  "baz",
]
`;

exports[`10 add  1`] = `
Array [
  "bar",
  "baz",
]
`;
